site/content/how-to/traffic-management/snippets-filters.md

```diff
@@ -0,0 +1,345 @@
+---
+title: "Snippets Filters"
+weight: 800
+toc: true
+docs: "DOCS-000"
+---
+
+Learn how to use Snippets with the `SnippetsFilter` API.
+
+## Overview
+
+Snippets allow Cluster Operators to insert NGINX configuration into different contexts of the
+NGINX configurations that NGINX Gateway Fabric generates.
+
+Snippets are for advanced NGINX users who need more control over the generated NGINX configuration,
+and can be used in cases where Gateway API resources or NGINX extensions don't apply.
+
+## Disadvantages of Snippets
+
+Snippets are configured using the `SnippetsFilter` API, but are disabled by default due to their complexity and security implications.
+
+To use Snippets, set the `nginxGateway.snippetsFilters.enable` command line argument to true.
+
+Snippets have the following disadvantages:
+
+- *Complexity*. Snippets require you to:
+  - Understand NGINX Configuration primitives and implement a correct NGINX configuration.
+  - Understand how NGINX Gateway Fabric generates NGINX configuration so that a Snippet doesn’t interfere with the other features in the configuration.
```

Comment:
Should this be explained anywhere? That if an NGINX directive inserted through a snippet already exists, it'll break?